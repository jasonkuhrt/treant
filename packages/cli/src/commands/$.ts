import { Command } from '@effect/cli';
import { Console, Effect } from 'effect';
import { generateCommand } from './generate/$.js';

export const mainCommand = Command.make('treant', {}, () =>
  Effect.gen(function*() {
    yield* Console.log('Treant CLI - Tree-sitter tools');
    yield* Console.log('Use "treant generate" to generate both grammar and SDK');
    yield* Console.log('Use "treant generate grammar" or "treant generate library" for specific tasks');
  })).pipe(Command.withSubcommands([generateCommand]));
