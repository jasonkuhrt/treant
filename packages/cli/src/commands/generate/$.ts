import { Command } from '@effect/cli';
import { Console, Effect } from 'effect';
import { libraryCommand } from './library.js';

export const generateCommand = Command.make('generate', {}, () =>
  Effect.gen(function*() {
    yield* Console.log('Available commands:');
    yield* Console.log('  treant generate library  - Generate TypeScript SDK from grammar');
  })).pipe(Command.withSubcommands([libraryCommand]));
