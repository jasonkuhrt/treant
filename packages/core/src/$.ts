export * as Core from './$$.js';
