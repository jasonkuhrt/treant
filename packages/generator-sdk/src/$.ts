export * as GeneratorSDK from './$$.js';
