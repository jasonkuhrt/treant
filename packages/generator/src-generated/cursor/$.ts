// Auto-generated from parser/src/node-types.json and parser/src/grammar.json
// DO NOT EDIT - This file is overwritten by scripts/gen-grammar-lib.ts
//
// Cursor navigation system namespace export
/**
 * Type-safe cursor navigation system.
 * 
 * This module provides exhaustive type-safe cursor navigation
 * with compile-time guarantees about AST structure.
 */
export * from "./$$.js";
