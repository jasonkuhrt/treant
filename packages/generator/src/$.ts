export * as Generator from './$$.js';
