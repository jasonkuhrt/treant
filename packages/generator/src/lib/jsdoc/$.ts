/**
 * JSDoc syntax building utilities.
 * 
 * This namespace provides utilities for programmatically generating
 * JSDoc comments with proper formatting and structure.
 * 
 * @example
 * ```typescript
 * import * as JSDoc from './lib/jsdoc/$.js';
 * 
 * const comment = JSDoc.build({
 *   description: 'Does something useful',
 *   params: [{ name: 'value', type: 'string', description: 'The input value' }],
 *   returns: { description: 'The processed result' }
 * });
 * ```
 */

export * from './builders.js';