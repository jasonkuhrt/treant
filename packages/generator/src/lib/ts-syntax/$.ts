/**
 * TypeScript syntax construction utilities.
 *
 * This namespace provides utilities for generating TypeScript code
 * programmatically, including naming conventions, import/export statements,
 * type declarations, and tree-sitter specific helpers.
 *
 * @example
 * ```typescript
 * import * as TS from './lib/ts-syntax/$.js';
 *
 * const interfaceName = TS.getNodeInterfaceName('field_definition');
 * const guardName = TS.getTypeGuardName('field_definition');
 *
 * const unionType = TS.unionType(['Field', 'Fragment', 'Operation']);
 * ```
 */

export * from './ts-syntax.js';
