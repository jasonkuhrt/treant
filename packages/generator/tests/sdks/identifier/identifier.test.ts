import { describe, expect, expectTypeOf, test } from 'vitest';
import { TreantIdentifier as Identifier } from './identifier.sdk.fix/$.js';

test('Parser', async () => {
  const ast = await Identifier.Parser.parse('hello');
  expect(Identifier.Node.isSourceFile(ast.rootNode)).toBe(true);
  expect(Identifier.Node.isIdentifier(ast.rootNode.firstChild)).toBe(true);
});

describe('Navigator', () => {
  test('.create', async () => {
    const source = `hello`;
    const ast = await Identifier.Parser.parse(source);
    const $ = await Identifier.Navigator.create(ast);

    expect($.node).toMatchSnapshot();
    expectTypeOf($.node).toEqualTypeOf<Identifier.Node.SourceFile>();

    expect($.node.text).toEqual(source);
    expectTypeOf($.node.text).toEqualTypeOf<string>();

    expectTypeOf($.child()).toEqualTypeOf<Identifier.Node.Identifier>();

    // @ts-expect-error
    $.child(0);
  });
});
