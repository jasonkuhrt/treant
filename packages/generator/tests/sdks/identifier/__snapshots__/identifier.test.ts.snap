// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigator > .create 1`] = `
Node {
  "0": 0,
  "_children": undefined,
  "_namedChildren": undefined,
  "id": 105280,
  "startIndex": 0,
  "startPosition": {
    "column": 0,
    "row": 0,
  },
  "tree": Tree {
    "0": 105280,
    "language": Language {
      "0": 104916,
      "fields": [
        null,
      ],
      "types": [
        ,
        "identifier",
        "source_file",
      ],
    },
    "textCallback": [Function],
  },
}
`;
